src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1454,7 +1454,13 @@ bool isContainingAssemblyInReferences(ISymbol s)
                             // a containing assembly, we treat them as in the current assembly for access purposes
                             return assemblyIsInReferences(s.ContainingAssembly ?? this.Assembly);
                         default:
-                            return assemblyIsInReferences(s.ContainingAssembly);
+                            var containingAssembly = s.ContainingAssembly;
+                            if (containingAssembly is null)
+                            {
+                                Debug.Assert(false);
```

Comment:
:memo: I looked at implementations of `ContainingAssembly` that could return `null` and couldn't find any that isn't in a symbol kind addressed above.